core/auth/src/main/java/software/amazon/awssdk/auth/credentials/InstanceProfileCredentialsProvider.java

```diff
@@ -186,19 +201,25 @@ private RefreshResult<AwsCredentials> refreshCredentials() {
         } catch (Ec2MetadataClientException e) {
             if (e.statusCode() == 404) {
                 log.debug(() -> "Resolved profile is no longer available. Resetting it and trying again.");
-                resolvedProfile = null;
 
                 if (apiVersion == ApiVersion.UNKNOWN) {
                     apiVersion = ApiVersion.LEGACY;
                     return refreshCredentials();
+                } else if (resolveProfileName() == null) {
+                    // Resolved profile name is invalid, reset it and try again
+                    resolvedProfile = null;
+                    
+                    profileRetryCount++;
+                    if (profileRetryCount <= MAX_PROFILE_RETRIES) {
+                        log.debug(() -> "Profile name not found, retrying fetching the profile name again.");
+                        return refreshCredentials();
+                    }
+                } else {
+                    throw SdkClientException.builder()
+                                           .message("Invalid profile name")
+                                           .cause(e)
+                                           .build();
```

Comment:
Can we expand on this error message a bit? Specifically, what's the invalid value and what should they do about it?